src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -294,11 +378,48 @@ public DetailedPackageMetadata PackageMetadata
                 if (_packageMetadata != value)
                 {
                     _packageMetadata = value;
+
+                    if (_packageMetadata?.DeprecationMetadata != null)
+                    {
+                        PackageDeprecationReasons = ExplainPackageDeprecationReasons(_packageMetadata.DeprecationMetadata.Reasons?.ToList());
+                        PackageDeprecationCustomMessage = string.IsNullOrWhiteSpace(_packageMetadata.DeprecationMetadata.Message) ? null : _packageMetadata.DeprecationMetadata.Message;
+
+                        var alternatePackage = _packageMetadata.DeprecationMetadata.AlternatePackage;
+                        if (alternatePackage != null)
+                        {
+                            PackageDeprecationAlternatePackageText = GetPackageDeprecationAlternatePackageText(alternatePackage);
+                        }
+                    }
+
                     OnPropertyChanged(nameof(PackageMetadata));
+                    OnPropertyChanged(nameof(IsPackageDeprecated));
                 }
             }
         }
 
+        private string GetPackageDeprecationAlternatePackageText(AlternatePackageMetadata alternatePackageMetadata)
+        {
+            if (alternatePackageMetadata == null)
+            {
+                return null;
+            }
+
+            string text = alternatePackageMetadata.PackageId;
+
+            if (!alternatePackageMetadata.Range.IsFloating
+                && alternatePackageMetadata.Range.HasLowerBound)
+            {
+                text = $"{alternatePackageMetadata.PackageId} version {alternatePackageMetadata.Range.MinVersion.ToNormalizedString()}";
```

Comment:
Use a format string if this is going to be use facing. 

Shouldn't it be localized as well?

Reply:
Fixed in https://github.com/NuGet/NuGet.Client/pull/2899/commits/89fc06d5415116f0f5265e809a60c3a21bead04f